File: src/main/java/com/korealm/emanon/auth/internal/data/models/LoginHistory.java
package com.korealm.emanon.auth.internal.data.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.net.InetAddress;
import java.time.OffsetDateTime;

@Getter
@Setter
@Entity
@Table(name = "login_history", schema = "auth")
public class LoginHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "success", updatable = false, nullable = false)
    private Boolean success;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "attempt_at", insertable = false, updatable = false, nullable = false)
    private OffsetDateTime attemptAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @OnDelete(action = OnDeleteAction.RESTRICT)
    @JoinColumn(name = "user_id", updatable = false)
    private AppUser user;

    @Column(name = "ip_address", updatable = false)
    private InetAddress ipAddress;

    @Column(name = "user_agent", updatable = false, nullable = false)
    private String userAgent;


}

File: src/main/kotlin/com/korealm/emanon/LoginHistory.kt
package com.korealm.emanon

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.ColumnDefault
import java.net.InetAddress
import java.time.OffsetDateTime

@Entity
@Table(name = "login_history", schema = "auth")
class LoginHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    var id: Long = 0L

    @Column(name = "user_id", nullable = false)
    var userId: Long = 0L

    @Column(name = "ip_address", nullable = false)
    var ipAddress: InetAddress? = null

    @Column(name = "user_agent", nullable = false)
    var userAgent: String = ""

    @Column(name = "success", nullable = false)
    var success: Boolean = false

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "attempt_at", nullable = false)
    var attemptAt: OffsetDateTime = OffsetDateTime.now()

}